tests/test_csc.py

```diff
@@ -0,0 +1,302 @@
+            self.assertEqual(state.npixels, self.patcher.n_pixels)
+            self.assertEqual(state.fpgaVersion, self.patcher.fpga_version)
+            self.assertEqual(state.firmwareVersion, self.patcher.firmware_version)
+            self.assertEqual(state.libraryVersion, self.patcher.library_version)
+
+            # In ENABLED, there should still be a connected device.
+            await harness.remote.cmd_enable.start(timeout=STD_TIMEOUT)
+            await self.check_summaryState(harness.remote, salobj.State.ENABLED)
+            self.assertIsNotNone(harness.csc.device)
+            # we only should try to activate once in this whole sequence
+            self.patch.return_value.AVS_Activate.assert_called_once()
+
+            # In DISABLED, there should still be a connected device.
+            await harness.remote.cmd_disable.start(timeout=STD_TIMEOUT)
+            await self.check_summaryState(harness.remote, salobj.State.DISABLED)
+            self.assertIsNotNone(harness.csc.device)
+            # we only should try to activate once in this whole sequence
+            self.patch.return_value.AVS_Activate.assert_called_once()
+
+            # in STANDBY, there should be no connected device
+            await harness.remote.cmd_standby.start(timeout=STD_TIMEOUT)
+            await self.check_summaryState(harness.remote, salobj.State.STANDBY)
+            self.assertIsNone(harness.csc.device)
+            # Entering standby should close the connection; Done might be
+            # called twice: once for the explicit `disconnect()` and  once
+            # for the destruction of the device object (but due to python
+            # garbage collection, the latter is not guaranteed to have happened
+            # immediately).
+            self.patch.return_value.AVS_Done.assert_called()
+
+    async def test_enable_fails(self):
+        """Test that exceptions raised when connecting cause a fault when
+        switching the CSC from STANDBY to DISABLED.
+        """
+        self.patch.return_value.AVS_Activate.return_value = AvsReturnCode.invalidHandle.value
+        async with Harness(initial_state=salobj.State.STANDBY) as harness:
+            # Check that we are properly in STANDBY at the start
+            await self.check_summaryState(harness.remote, salobj.State.STANDBY)
+
+            msg = "Failed to connect"
+            with salobj.assertRaisesAckError(ack=salobj.SalRetCode.CMD_FAILED, result_contains=msg):
+                await harness.remote.cmd_start.start(timeout=STD_TIMEOUT)
+            state = await harness.remote.evt_summaryState.next(flush=False, timeout=STD_TIMEOUT)
+            self.assertEqual(state.summaryState, salobj.State.FAULT)
+            error = await harness.remote.evt_errorCode.next(flush=False, timeout=STD_TIMEOUT)
+            self.assertIn("RuntimeError('Invalid device handle; cannot activate device", error.errorReport)
```

Comment:
Consider a less detailed test, such as checking that "RuntimeError" is in the string and also that some other text, such as "Invalid device handle" is in the string.